lib/kafka_gen_stage/consumer_logic.ex

```diff
@@ -80,4 +92,11 @@ defmodule KafkaGenStage.ConsumerLogic do
     |> Stream.take_while(fn {offset, _, _, _} -> offset <= end_offset end)
     |> Enum.reduce(queue, &:queue.in/2)
   end
+
+  defp dequeue_eos_offsets(queue, until) do
+    case :queue.peek(queue) do
+      {:value, offset} when offset <= until -> dequeue_eos_offsets(:queue.tail(queue), until)
+      _ -> queue
+    end
+  end
```

Comment:
What about moving the dequeue of the eos queue into the branch when the last_offset is real? In the other branch, let's just keep the eos queue as is. It could help readability.

Reply:
Yeah, I'll move it as you suggest.

Reply:
Fixed in ebd06f6